ksqldb-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -407,6 +407,15 @@
       "Feature flag for ROWPARTITION and ROWOFFSET pseudocolumns. If enabled, new queries will be"
           + "built with ROWPARTITION and ROWOFFSET pseudocolumns. If off, they will not be.";
 
+  public static final String KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED =
+      "ksql.source.table.materialization.enabled";
+  private static final Boolean KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED_DEFAULT = true;
+  private static final String KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED_DOC =
+      "Feature flag that enables table materialization on source tables. "
+          + "A source table is created with the CREATE SOURCE TABLE syntax. "
+          + "Default is true. If false, CREATE SOURCE TABLE will not be materialized, "
```

Comment:
if the flag is false, it's more accurate to say that CST statements would be rejected, right?

Reply:
Yes, I'll change this. I initially was going to disable materialization only, and forgot to edit this part.